offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/offers/OfferDbConnector.groovy

```diff
@@ -320,9 +320,16 @@ class OfferDbConnector implements CreateOfferDataSource, FetchOfferDataSource, P
             return identifier
         }
         try {
+            /*
+            A full openBIS project ID has the format: '/<space>/<project>', where
+            <space> and <project> are placeholders for real space and project names.
+             */
             def splittedIdentifier = projectIdentifier.split("/")
-            def space = new ProjectSpace(splittedIdentifier[0])
-            def code = new ProjectCode(splittedIdentifier[1])
+            if (splittedIdentifier.length != 3) {
+                throw new RuntimeException("Project identifier has an unknown format: ${projectIdentifier.split()}.")
```

Comment:
I like that a RuntimeException is thrown :+1: You could be even more specific since we do not really check the whole format here but only the number of slashes. What is `projectIdentifier.split()` doing here? I think the default split would be a whitespace character? Wouldn't we want the `splittedIdentifier` in the error message?
```suggestion
                throw new RuntimeException("Project identifier has an unexpected number of separators: ${projectIdentifier.split()}.")
```
```suggestion
                throw new RuntimeException("Project identifier has an unexpected number of separators: ${splittedIdentifier}.")
```